src/transformers/trainer.py

```diff
@@ -1817,14 +1840,34 @@ def _load_from_checkpoint(self, resume_from_checkpoint, model=None):
             # will be resumed in deepspeed_init
             pass
         elif os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME)):
-            # We load the model state dict on the CPU to avoid an OOM error.
-            state_dict = torch.load(os.path.join(resume_from_checkpoint, WEIGHTS_NAME), map_location="cpu")
             # If the model is on the GPU, it still works!
-            load_result = model.load_state_dict(state_dict, strict=strict_load)
+            if strict_load:
```

Comment:
Had to dig into the code to realise `strict_load` is just `is_sagemaker_mp_enabled()`. So let's remove the definition at line 1797/1820 and just use
```suggestion
            if is_sagemaker_mp_enabled():
```
here so the code is easier to read.